include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -372,108 +375,124 @@ namespace LinearAlgebra
-          ::dealii::Utilities::CUDA::malloc(V_dev, n_elements);
-          cudaError_t cuda_error_code = cudaMemcpy(V_dev,
-                                                   V.begin(),
-                                                   n_elements * sizeof(Number),
-                                                   cudaMemcpyHostToDevice);
-          AssertCuda(cuda_error_code);
+          Kokkos::View<Number *, Kokkos::HostSpace> V_view(V.begin(),
+                                                           n_elements);
+          auto V_dev = Kokkos::create_mirror_view_and_copy(
+            ::dealii::MemorySpace::Default::kokkos_space{}, V_view);
 
           // Set the values in tmp_vector
-          const int n_blocks =
-            1 + n_elements / (::dealii::CUDAWrappers::chunk_size *
-                              ::dealii::CUDAWrappers::block_size);
-          ::dealii::LinearAlgebra::CUDAWrappers::kernel::set_permutated<Number>
-            <<<n_blocks, ::dealii::CUDAWrappers::block_size>>>(
-              indices_dev, tmp_vector.begin(), V_dev, n_elements);
+          Kokkos::parallel_for(
+            "import_elements: set values tmp_vector",
+            Kokkos::RangePolicy<
+              ::dealii::MemorySpace::Default::kokkos_space::execution_space>(
+              exec, 0, n_elements),
+            KOKKOS_LAMBDA(size_type i) {
+              tmp_vector(indices_dev(i)) = V_dev(i);
+            });
+          exec.fence();
 
           tmp_vector.compress(operation);
 
           // Copy the local elements of tmp_vector to the right place in val
           IndexSet        tmp_index_set  = tmp_vector.locally_owned_elements();
           const size_type tmp_n_elements = tmp_index_set.n_elements();
-          indices.resize(tmp_n_elements);
-          for (size_type i = 0; i < tmp_n_elements; ++i)
-            indices[i] = locally_owned_elem.index_within_set(
-              tmp_index_set.nth_index_in_set(i));
-          ::dealii::Utilities::CUDA::free(indices_dev);
-          ::dealii::Utilities::CUDA::malloc(indices_dev, tmp_n_elements);
-          ::dealii::Utilities::CUDA::copy_to_dev(indices, indices_dev);
+          Kokkos::realloc(indices, tmp_n_elements);
+          Kokkos::parallel_for(
+            "import_elements: copy local elements to val",
+            Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(host_exec,
+                                                                   0,
+                                                                   n_elements),
+            KOKKOS_LAMBDA(size_type i) {
+              indices[i] = locally_owned_elem.index_within_set(
+                tmp_index_set.nth_index_in_set(i));
+            });
+          host_exec.fence();
+          Kokkos::realloc(indices_dev, tmp_n_elements);
+          Kokkos::deep_copy(indices_dev,
+                            Kokkos::View<size_type *>(indices.data(),
+                                                      tmp_n_elements));
```

Comment:
`indices` is already a `View` so why do you create a new one? Shouldn't this be a subview?

Reply:
Fixed.